src/ontology/mondo-ingest.Makefile

```diff
@@ -369,44 +367,43 @@ deploy-mondo-ingest:
 	ls -alt $(DEPLOY_ASSETS_MONDO_INGEST)
 	gh release create $(GHVERSION) --notes "TBD." --title "$(GHVERSION)" --draft $(DEPLOY_ASSETS_MONDO_INGEST)
 
-USE_MONDO_RELEASE=false
 
+# make function, not target! 
 # Builds tmp/mondo/ and rebuilds mondo.owl and mondo.sssom.tsv, and stores hash of latest commit of mondo repo main branch in tmp/mondo_repo_built
-tmp/mondo_repo_built:
-	if [ $(USE_MONDO_RELEASE) = true ]; then wget http://purl.obolibrary.org/obo/mondo.owl -O $@; else cd $(TMPDIR) &&\
-		rm -rf ./mondo/ &&\
-		git clone --depth 1 https://github.com/monarch-initiative/mondo &&\
-		cd mondo/src/ontology &&\
-		make mondo.owl mappings -B MIR=false IMP=false MIR=false &&\
-		latest_hash=$$(git rev-parse origin/master) &&\
-		echo "$$latest_hash" > $@ &&\
-		cp $@ mappings/mondo.sssom.tsv mondo.owl ../../../; fi
-
-# Triggers a refresh of tmp/mondo/ and a rebuild of mondo.owl and mondo.sssom.tsv, only if mondo repo main branch has new commits
-.PHONY: refresh-mondo-clone
-refresh-mondo-clone:
-	@if [ ! -d tmp/mondo ]; then \
-		$(MAKE) tmp/mondo_repo_built -B; \
+
+define build_mondo
+	cd $(TMPDIR) && \
+	rm -rf ./mondo/ && \
+	git clone --depth 1 https://github.com/monarch-initiative/mondo && \
+	cd mondo/src/ontology && \
+	make mondo.owl mappings -B MIR=false IMP=false MIR=false \
+	latest_hash=$$(git rev-parse origin/master) && \
+	cd ../../../.. && \
+	echo "$$latest_hash" > $(1)
+endef
+
+tmp/mondo_repo_built: .FORCE
+	@if [ ! -f $@ ]; then \
+		$(call build_mondo, $@); \
 	else \
-		current_hash=$$(cat tmp/mondo_repo_built); \
+		current_hash=$$(cat $@); \
 		cd tmp/mondo; \
 		git fetch origin; \
 		latest_hash=$$(git rev-parse origin/master); \
 		if [ ! "$$current_hash" = "$$latest_hash" ]; then \
 			cd .. && mv mondo mondo-bak && mv mondo_repo_built mondo_repo_built-bak; \
-			cd .. && $(MAKE) tmp/mondo_repo_built -B; \
+			cd .. && $(call build_mondo, $@); \
```

Comment:
@joeflack4 this might interest you. A heavily underused `make` pattern! `make` functions! Cool!

Reply:
You're right, it does. I only tried them once or twice and had difficulty getting the params to work. In this age of LLM I believe I'd have a much easier time.

There will be no variation in params for this function, so my guess is that it is effectively the same and there is no advantage/difference other than style, but if there is more to it, lemme know.